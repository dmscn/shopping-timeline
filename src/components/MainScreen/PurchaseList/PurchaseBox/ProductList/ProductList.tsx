import React from "react";
import { DataTable, Box, Text } from "grommet";

export interface Props {
  products: [];
}

export default function ProductList(props: Props): JSX.Element {
  const { products } = props;

  return (
    <DataTable
      columns={[
        {
          property: "product_name",
          header: <Text>Produto</Text>,
          primary: true
        },
        {
          property: "revenue",
          header: <Text>Pre√ßo</Text>
        }
      ]}
      data={products}
    />
  );
}
