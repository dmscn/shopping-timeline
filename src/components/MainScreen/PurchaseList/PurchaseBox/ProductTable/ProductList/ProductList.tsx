import React from "react";
import { TableRow, TableCell } from "grommet";

export interface Props {
  products: Array<any>;
}

export default function ProductList(props: Props): any {
  const result: Array<JSX.Element> = props.products.map((product: any) => {
    const { value: productName } = product.custom_data.find(
      (item: any) => item.key === "product_name"
    );
    const { value: productPrice } = product.custom_data.find(
      (item: any) => item.key === "product_price"
    );

    return (
      <TableRow key={product.timestamp}>
        <TableCell scope="row">{productName}</TableCell>
        <TableCell>{productPrice}</TableCell>
      </TableRow>
    );
  });
  return result;
}
