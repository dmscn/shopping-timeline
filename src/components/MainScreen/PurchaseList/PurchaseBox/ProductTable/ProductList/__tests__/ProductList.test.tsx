import React from "react";
import { shallow } from "enzyme";

import ProductList from "../ProductList";

describe("<ProductList />", () => {
  it("renders", () => {
    const wrapper = shallow(<ProductList products={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
