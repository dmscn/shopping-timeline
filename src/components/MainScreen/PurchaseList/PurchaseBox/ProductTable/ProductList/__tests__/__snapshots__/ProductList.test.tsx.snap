// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductList /> renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductList
    products={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): undefined,
  Symbol(enzyme.__nodes__): Array [],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
